modules/gdscript/gdscript_compiler.cpp

```diff
@@ -1373,6 +1375,62 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 
 			return result;
 		} break;
+		case GDScriptParser::Node::COMPREHENSION: {
+			const GDScriptParser::ComprehensionNode *comprehension = static_cast<const GDScriptParser::ComprehensionNode *>(p_expression);
+			// std::cout << "adding result temporary" << std::endl;
+			GDScriptCodeGenerator::Address result = codegen.add_temporary(_gdtype_from_datatype(comprehension->get_datatype(), codegen.script));
+			// std::cout << "added result temporary" << std::endl;
+
+			Vector<GDScriptCodeGenerator::Address> values;
+
+			gen->write_construct_array(result, values);
+
+			codegen.start_block();
+			GDScriptCodeGenerator::Address iterator = codegen.add_local(comprehension->variable->name, _gdtype_from_datatype(comprehension->variable->get_datatype(), codegen.script));
+
+			gen->start_for(iterator.type, _gdtype_from_datatype(comprehension->list->get_datatype(), codegen.script));
+
+			GDScriptCodeGenerator::Address list = _parse_expression(codegen, r_error, comprehension->list);
+			if (r_error) {
+				return GDScriptCodeGenerator::Address();
+			}
+
+			gen->write_for_assignment(list);
+
+			if (list.mode == GDScriptCodeGenerator::Address::TEMPORARY) {
+				codegen.generator->pop_temporary();
+			}
+
+			gen->write_for(iterator, false); // ?: is this false correct?
+
+			GDScriptCodeGenerator::Address expression = _parse_expression(codegen, r_error, comprehension->expression);
+			if (r_error) {
+				return GDScriptCodeGenerator::Address();
+			}
+			values.push_back(expression);
+			gen->write_call_builtin_type(GDScriptCodeGenerator::Address(), result, Variant::Type::ARRAY, "push_back", values);
+
+			// if (result.mode == GDScriptCodeGenerator::Address::TEMPORARY) {
+			// 	codegen.generator->pop_temporary();
+			// }
+
+			for (auto &e : static_cast<GDScriptByteCodeGenerator *>(codegen.generator)->used_temporaries) {
```

Comment:
```suggestion
			for (int e : static_cast<GDScriptByteCodeGenerator *>(codegen.generator)->used_temporaries) {
```
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)

Reply:
This is just used for debug purposes, will be removed in the final pr (see the inclusion of iostream and the usage of cout)

Reply:
Use `print_line` for debugging instead :)

Reply:
But do please follow the rules for even draft and temporary stuff, it's good to learn to write code that's in line with how code is expected to be